src/user.c

```diff
@@ -40,6 +40,16 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 
 extern int eos;
 
+/** Lookup an account from a Nick 
+ * @param   nick    pointer to a Nick
+ * @returns an User struct
+ * @note    This is an helper function to ease the migration to an account based 
+ *          paradigm.
+ */
+User *find_account(Nick *nick) {
+    return nick->svid ? find_user(nick->svid) : find_user(nick->nick);
```

Comment:
This should rather be:

return *nick->svid ? ...

Note the dereference on svid, checking that the first character is not '\0'. Otherwise, the expression "nick->svid" will always evaluate to true given that svid is an array.

Reply:
Fixed